File: src/main/java/com/woomulwoomul/woomulwoomulbackend/common/response/ExceptionResponse.java
package com.woomulwoomul.woomulwoomulbackend.common.response;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;
import org.springframework.http.ResponseEntity;

@Builder
@Schema(title = "* 예외 응답")
public record ExceptionResponse(
        @Schema(description = "코드", example = "RESPONSE_CODE")
        String code,
        @Schema(description = "메세지", example = "응답 메세지입니다.")
        String message
) {

    public static ResponseEntity<ExceptionResponse> ofResponseEntity(ExceptionCode exceptionCode) {
        return ResponseEntity.status(exceptionCode.getStatus().value())
                .body(ExceptionResponse.builder()
                        .code(exceptionCode.name())
                        .message(exceptionCode.getMessage())
                        .build());
    }
}

File: src/main/kotlin/com/woomulwoomul/woomulwoomulbackend/common/response/ExceptionResponse.kt
package com.woomulwoomul.woomulwoomulbackend.common.response

import com.woomulwoomul.woomulwoomulbackend.common.constant.ExceptionCode
import org.springframework.http.ResponseEntity

class ExceptionResponse(
    val code: String,
    val message: String,
) {
    constructor(exceptionCode: ExceptionCode) : this(exceptionCode.name, exceptionCode.message)

    companion object {
        fun toResponseEntity(ex: CustomException): ResponseEntity<ExceptionResponse> {
            return ResponseEntity.status(ex.exceptionCode.httpStatus)
                .body(ExceptionResponse(ex.exceptionCode.name, ex.exceptionCode.message))
        }

        fun toResponseEntity(exceptionCode: ExceptionCode): ResponseEntity<ExceptionResponse> {
            return ResponseEntity.status(exceptionCode.httpStatus)
                .body(ExceptionResponse(exceptionCode.name, exceptionCode.message))
        }
    }
}

class CustomException(
    val exceptionCode: ExceptionCode,
    val throwable: Throwable? = null,
) : RuntimeException()